docs/feature_autocorrect.md

```diff
@@ -0,0 +1,297 @@
+# Autocorrect
+
+There are a lot of words that are prone to being typed incorrectly, due to habit, sequence or just user error.  This feature leverages your firmware to automatically correct these errors, to help reduce typos.
+
+## How does it work? :id=how-does-it-work
+
+The feature maintains a small buffer of recent key presses. On each key press, it checks whether the buffer ends in a recognized typo, and if so, automatically sends keystrokes to correct it.
+
+The tricky part is how to efficiently check the buffer for typos. We don’t want to spend too much memory or time on storing or searching the typos. A good solution is to represent the typos with a trie data structure. A trie is a tree data structure where each node is a letter, and words are formed by following a path to one of the leaves.
+
+![An example trie](https://getreuer.info/posts/keyboards/autocorrection/trie.svg)
```

Comment:
In case my personal website ever goes away, it would be more future proof to add this image in QMK under docs/. The image is small (12 KB) and there are a couple other SVGs under docs/ already, so I hope no problem to include it.

Reply:
Ya, it's reasonable to have a rule like this about image hosting. My understanding is the concern is inflating the repo size, imagining if many users uploaded multi-megabyte photos of their keyboards and PCBs. Maybe this trie.svg is small enough to make an exception. [Here is a PNG conversion](https://github.com/getreuer/getreuer.github.io/commit/e341a1e3e9dafa259214719df8c0bbe1be78a465) of trie.svg in case it's useful to host on imgur or elsewhere.